### DISCUSSION TITLE
Benchmarking gnark and bellman

### DISCUSSION THREAD

**Discussion by sshravan** [Consensys/gnark#330] [Q&A]
Hello, 
I am trying to understand the performance of gnark on my target machine. 

Is the code used to report the performance numbers in the [benchmarks page](https://docs.gnark.consensys.net/en/latest/#gnark-is-fast) available? 

I noticed the [gnark-bench repo](https://github.com/ConsenSys/gnark-bench). But, I am unsure how the bellman numbers were obtained in the benchmarks page. Any pointers regarding this is appreciated!

**Comment by gbotrel** [comment]
hi -- updated the gnark-bench to use latest gnark version. The bench circuit is here: https://github.com/ConsenSys/gnark-bench/blob/3d78a9afb71aadc9d7302ab56c3ebceec2cb90cb/circuit/circuit.go#L105 . 

bellman numbers were obtain with the same circuit and size in bellman -- If I recall correctly it's a slight modification to the bellman benchmark (cc @yelhousni ) . 

Example command for gnark-bench; 
```
go build -tags=amd64_adx [if on x86 machine]
./gnark-bench groth16 --size 1000000 --count 5
```

**Comment by sshravan** [comment]
Thank you for updating [```gnark-bench```](https://github.com/ConsenSys/gnark-bench)! 

Is there a ```bellman-bench``` repo that was used to report the numbers in the benchmark page? 

I am trying to figure out the parameters/flags that were used for bellman numbers (```asm```, ```BELLMAN_NO_GPU```, etc).

**Comment by gbotrel** [comment]
there is no "bellman-bench" repo to the best of my knowledge. If you do some work to compare various zkSNARK libraries, please let us know :-)
